comp110-worksheet-6/OxoBoard.cs

```diff
@@ -4,73 +4,196 @@
-						case Mark.None:
-							Console.Write(" "); break;
-						case Mark.O:
-							Console.Write("O"); break;
-						case Mark.X:
-							Console.Write("X"); break;
-					}
-				}
-
-				Console.WriteLine();
-			}
-		}
-	}
-}
+    public enum Mark { None, O, X };
+
+
+    public class OxoBoard
+    {
+        Mark[,] board;
+        int width = 3;
+        int height = 3;
+        int inARow = 3;
+
+        // Constructor. Perform any necessary data initialisation here.
+        // Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
+
+        //Initialize the board with a 2D array and then iterate through each one of its coordinates, marking them as empty/none therefore creating an "empty" board.
+        public OxoBoard(int width = 3, int height = 3, int inARow = 3)
+        {
+            board = new Mark[3, 3];
+
+            for (int i = 0; i < width; i++)
+            {
+                for (int j = 0; j < height; j++)
+                    board[i, j] = Mark.None;
+            }
+        }
+
+        // Return the contents of the specified square.
+        public Mark GetSquare(int x, int y)
+        {
+            return board[x, y];
+        }
+        // If the specified square is currently empty, fill it with mark and return true.
+        // If the square is not empty, leave it as-is and return False.
+
+            //Check if the coordinates, x and y, don't surpass the boundaries.
+
+        public bool SetSquare(int x, int y, Mark mark)
+        {
+            int counter;
+
+            if ((x >= 0 && x < 3) && (y >= 0 && y <3))
+            {
+                if (board[x, y] == Mark.None)
+                {
+                    board[x, y] = mark;
+                    return true;
+                }
+            }
+            return false;
+
+        }
+
+        //counter++;
+
+
+        // If there are still empty squares on the board, return false.
+        // If there are no empty squares, return true.
+
+        //Iterate through each square and check if it is empty. If it it is, the board is not full, therefore return false. Otherwise, return true after finish iterating through each one of them.
+        public bool IsBoardFull()
+        {
+
+
+            for (int i = 0; i < width; i++)
+                for (int j = 0; j < height; j++)
+                    if (board[i, j] == Mark.None)
+                        return false;
+            return true;
+
+
+
+        }
+
+        // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        // Otherwise, return Mark.None.
+        public Mark GetWinner()
+        {
+            //Check for vertical marks + making sure the square that is in a row is not empty
+            for (int i = 0; i < 3; i++)
+            {
+                if (board[i, 0] == board[i, 1] && board[i, 1] == board[i, 2] && GetSquare(i, 0) != Mark.None)
+                    return GetSquare(i, 0);
```

Comment:
You're using both `board[]` and `GetSquare()` in this function -- either is fine but be consistent